core/shared/src/main/scala/fs2/Stream.scala

```diff
@@ -4196,6 +4196,37 @@ object Stream extends StreamLowPriority {
     }
   }
 
+  /** Provides syntax for list of streams. */
+  implicit final class ListStreamOps[F[_], O](private val xs: List[Stream[F, O]]) extends AnyVal {
+
+    /** Nondeterministically merges a (static) list of streams in to a single output stream.
```

Comment:
Question: should
```scala
List(a, b, c).parJoinUnbounded
```
be semantically equivalent to
```scala
a.merge(b).merge(c)
```
?

If not, what's the difference?

If so, then I think the implementation should look more like `merge`.

<details>

https://github.com/typelevel/fs2/blob/f6702f0de74202af7272874ce3a18d0741f8cca7/core/shared/src/main/scala/fs2/Stream.scala#L1904-L2005

</details>

Also, in that case I wonder if `parJoinUnbounded` is the right name. What if we exposed this API like:
```scala
object Stream {
  def merge[F[_], A](streams: Stream[F, A]*): Stream[F, A] = ???
}
```

or `mergeAll` or `mergeMany` or something?

Reply:
That is a good question, I would go back one step and ask if
```scala
Stream(a, b).parJoinUnbounded
```
is semantically equivalent to
```scala
a.merge(b)
```
?
Citing the `merge` scaladoc:
```
Note that even when this is equivalent to `Stream(this, that).parJoinUnbounded`, this implementation is little more efficient
```
Looking at the implementation, I am not sure, if it is equivalent. `merge` adds some fairness with regard to chunks? (Also, as for the performance, simple benchmarks hint that the opposite is actually true - `merge` is slower compared to `parJoinUnbounded`).